src/whir/mod.rs

```diff
@@ -16,39 +15,30 @@ pub mod verifier;
 
 // Only includes the authentication paths
 #[derive(Clone, Debug, CanonicalSerialize, CanonicalDeserialize, Serialize, Deserialize)]
-pub struct WhirProof<MerkleConfig, F>
+pub struct WhirProof<F>
 where
-    MerkleConfig: Config<Leaf = [F]>,
     F: Sized + Clone + CanonicalSerialize + CanonicalDeserialize,
 {
-    #[serde(with = "crate::ark_serde")]
-    pub merkle_paths: Vec<(MultiPath<MerkleConfig>, Vec<Vec<F>>)>,
     #[serde(with = "crate::ark_serde")]
     pub statement_values_at_random_point: Vec<F>,
```

Comment:
I don't think this should be here either, but I'll leave it for a follow up PR to remove. I think it's the verifiers responsibility to compute this.

Reply:
@veljkovranic Why is `statement_values_at_random_point` part of the proof? Shouldn't verifier compute this?

Reply:
I moved these to verifier work, but for a practical verifier I'll have to re-add a method to defer or precompute the evaluation and not force a full evaluation of the MLE in the verifier.